packages/mermaid/src/diagrams/gantt/ganttDb.js

```diff
@@ -343,13 +337,36 @@ const parseDuration = function (str) {
 const getEndDate = function (prevTime, dateFormat, str, inclusive = false) {
   str = str.trim();
 
-  // Check for actual date
-  let mDate = dayjs(str, dateFormat.trim(), true);
-  if (mDate.isValid()) {
+  // test for until
+  const untilRePattern = /^until\s+(?<ids>[\d\w- ]+)/;
+  const untilStatement = untilRePattern.exec(str);
+
+  if (untilStatement !== null) {
+    // check all until ids and take the earliest
+    let earliestTask = null;
+    for (const id of untilStatement.groups.ids.split(' ')) {
+      let task = findTaskById(id);
+      if (task !== undefined && (!earliestTask || task.startTime < earliestTask.startTime)) {
+        earliestTask = task;
+      }
+    }
+
+    if (earliestTask) {
+      return earliestTask.startTime;
+    } else {
+      const today = new Date();
+      today.setHours(0, 0, 0, 0);
+      return today;
+    }
```

Comment:
Minor styling suggestion. Here and above for `latestTask`.
```suggestion
    if (earliestTask) {
      return earliestTask.startTime;
    }
    const today = new Date();
    today.setHours(0, 0, 0, 0);
    return today;
```